### ISSUE TITLE
[BUG] Ensure CompletableFuture completion in HttpMcpTransport

### ISSUE THREAD

**Issue by andreadimaio** [langchain4j/langchain4j#2341]
I'm looking at the [HttpMcpTransport](https://github.com/langchain4j/langchain4j/blob/e62db38df052702b4ab3170a3c0b214e4dab2075/langchain4j-mcp/src/main/java/dev/langchain4j/mcp/client/transport/http/HttpMcpTransport.java#L125-L144) implementation and it seems that the `onResponse` method never sets the `future.complete(...)`. A possible solution could be:
```java
@Override
public void onResponse(Call call, Response response) throws IOException {
    int statusCode = response.code();
    if (!isExpectedStatusCode(statusCode)) {
        future.completeExceptionally(new RuntimeException("Unexpected status code: " + statusCode));
        return;
    }

    try {
        String responseBody =
            response.body() != null ? response.body().string() : null;
        if (responseBody == null || responseBody.isEmpty()) {
            future.completeExceptionally(new RuntimeException("Empty response body"));
            return;
        }
        JsonNode jsonNode = OBJECT_MAPPER.readTree(responseBody);
        future.complete(jsonNode);
    } catch (IOException e) {
        future.completeExceptionally(e);
    }
}
```

@jmartisk @langchain4j

**Comment by andreadimaio** [comment]
It also seems that the `Initialized` step is missing, which should be sent after the `Initialization` step as reported in the [documentation](https://spec.modelcontextprotocol.io/specification/basic/lifecycle/#initialization) .
> After successful initialization, the client MUST send an initialized notification to indicate it is ready to begin normal operations:
> ```json
> {
>  "jsonrpc": "2.0",
>  "method": "notifications/initialized"
>}
>```

**Comment by jmartisk** [comment]
No, obtaining a response from the POST endpoint doesn't yet mean that the operation has finished (it only means it was accepted for processing), completion happens when we receive a message with the right ID on the SSE stream. The `complete` method is called here: https://github.com/langchain4j/langchain4j/blob/e62db38df052702b4ab3170a3c0b214e4dab2075/langchain4j-mcp/src/main/java/dev/langchain4j/mcp/client/transport/McpOperationHandler.java#L29

As for the `notifications/initialized` bug, I agree, I'll fix it

**Comment by andreadimaio** [comment]
Ok, the problem is that the mpc server doesn't return the responses as an SSE event, or at least it doesn't in my local environment. This is the reason why the `initialization` does not "run" correctly.

**Comment by jmartisk** [comment]
@andreadimaio , I've submitted https://github.com/langchain4j/langchain4j/pull/2360 for sending the initialization notification

**Comment by andreadimaio** [comment]
I've tested it in my local environment and it seems to work fine. Now I'm able to close the `initialization/initialized` part!
There are changes to do on the mpc-server side to return the response via SSE events, but from the client side it looks good, thanks!

**Comment by langchain4j** [comment]
Should we close this PR or something is still missing?

**Comment by andreadimaio** [comment]
Resolved by #2360
